packages/runtime/container-runtime/src/dataStores.ts

```diff
@@ -611,15 +619,34 @@ export class DataStores implements IDisposable {
     }
 
     /**
-     * Returns the package path of the node with the given path. This is used by GC to log when an inactive / deleted
-     * node is used.
+     * Called by GC to retrieve the package path of the node with the given path. This is used log when an inactive or
+     * deleted node is used. The node should belong to a data store or be a blob.
      */
     public getNodePackagePath(nodePath: string): readonly string[] | undefined {
-        // Currently, only return the data store package path for the node since GC is only interested in data stores.
-        const dataStoreId = nodePath.split("/")[1];
-        const context = this.contexts.get(dataStoreId);
-        assert(context !== undefined, 0x2b9 /* "Data store with given id does not exist" */);
-        return context.isLoaded ? context.packagePath : undefined;
+        const pathParts = nodePath.split("/");
+        // If the node is a blob, return "_blobs" as the package path.
+        if (pathParts[1] === BlobManager.basePath) {
+            return ["_blobs"];
+        }
+
+        // If the node belongs to a data store, return its package path if the data store is loaded.
+        const context = this.contexts.get(pathParts[1]);
+        if (context !== undefined) {
+            return context.isLoaded ? context.packagePath : undefined;
+        }
+
+        assert(false, "Package path requested for unknown node type.");
+    }
+
+    /**
+     * Called by GC to know if a node is a data store or not. Data store ids are of the format "/dataStoreId".
+     */
+    public isDataStoreNode(nodePath: string): boolean {
+        const pathParts = nodePath.split("/");
+        if (pathParts.length === 2 && this.contexts.has(pathParts[1])) {
```

Comment:
other places exclude blob path. Might be worth centralizing such checks in one place.